packages/tailwind/src/tailwind.tsx

```diff
@@ -1,179 +1,142 @@
 import * as React from "react";
 import { renderToStaticMarkup } from "react-dom/server";
-import htmlParser, {
-  attributesToProps,
-  domToReact,
-  Element,
-} from "html-react-parser";
 import { tailwindToCSS, TailwindConfig } from "tw-to-css";
 
+import { cssToJsxStyle } from "./utils";
+
 export interface TailwindProps {
   children: React.ReactNode;
   config?: TailwindConfig;
 }
 
-export const Tailwind: React.FC<TailwindProps> = ({ children, config }) => {
+function processElement(
+  element: React.ReactElement,
+  headStyles: string[],
+  config?: TailwindConfig,
+): React.ReactElement {
   const { twi } = tailwindToCSS({
     config,
   });
 
-  const newChildren = React.Children.toArray(children);
-
-  const fullHTML = renderToStaticMarkup(<>{newChildren}</>);
+  if (element.props.className) {
+    const convertedStyles: string[] = [];
+    const responsiveStyles: string[] = [];
+    const classNames = element.props.className.split(" ");
+
+    const customClassNames = classNames.filter((className: string) => {
+      if (twi(className, { ignoreMediaQueries: true })) {
```

Comment:
```suggestion
      const tailwindClassName = twi(className, { ignoreMediaQueries: true })
      if (tailwindClassName) {
```

so we also reuse this in line 28